### PR TITLE
Adding a GitHub action to perform a Binary Diff job on each PR/push event

### PR DESCRIPTION
**Pull request by zeusongit**
### Purpose

Add a binary diff workflow that will run on each push and PR, this will enable devs to check for updated files (Added, Deleted or Modified) in their PR vs files on the master branch.

Note: caches will be removed automatically after 7 days.

Screenshot 2023-05-19 at 6 44 49 PM

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Release Notes

Add Bin Diff support on PRs

### Reviewers

@DynamoDS/dynamo

### CODE REVIEW

File: .github/scripts/bin_diff.ps1 [hunk 1]
```diff
@@ -0,0 +1,258 @@
+################################################################################
```
Comment (mjkkirschner) [inline .github/scripts/bin_diff.ps1:1]: is this script based on some sample code? Did it have a license, can we link to it?

File: .github/scripts/bin_diff.ps1 [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+# GETS ALL FILES WITH CALCULATED HASH PROPERTY RELATIVE TO A ROOT DIRECTORY RECURSIVELY #
+# RETURNS LIST OF @{RelativePath, Hash, FullName}
+function GetFilesWithHash {
+  [CmdletBinding()]
+  param (
+    [parameter(Mandatory=$TRUE,Position=0,HelpMessage="Path to get directories for.")]
+    [string]$Path,
+
+    [parameter(HelpMessage="The hash algorithm to use.")]
+    [string]$Algorithm="MD5"
```
Comment (mjkkirschner) [inline .github/scripts/bin_diff.ps1:154]: https://stackoverflow.com/questions/201705/how-many-random-elements-before-md5-produces-collisions
SHA256 perhaps?

File: .github/scripts/bin_diff.ps1 [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+
+
+### END FUNCTION DEFINITIONS ###
+
+### PROGRAM LOGIC ###
+
+if($Compare.length -ne 2) {
+  Print -x "Compare requires passing exactly 2 path parameters separated by comma, you passed $($Compare.length)." -f
+}
+Print "Comparing $($Compare[0]) to $($Compare[1])..." -a 1
```
Comment (mjkkirschner) [inline .github/scripts/bin_diff.ps1:226]: so I have a bit of a hard time understanding how this code works - but from your screenshot I have a few questions/comments.

1. I think noting that a file is `modified` if it still exists is probably just noise - IMO we don't really care, I expect files to be different when I do a build - they will contain different version numbers etc.
2. What I really care about are new files and deleted files - if we need to keep `modified` files - can we make deleted and added files pop out some how? (color text?)
3. Is a deleted file also considered modified or does it have a separate state? Can you show a screenshot of this?

File: .github/workflows/dynamoBinDiff.yml [hunk 4]
```diff
@@ -0,0 +1,86 @@
+# Build Dynamo using latest VS and DotNET and perform a Bin Diff
+name: Dynamo-BinDiff
+on: [push,pull_request]
+jobs:
+ build-current:
+  runs-on: windows-2022
+  steps:
+    - name: Checkout Dynamo Repo current branch
```
Comment (mjkkirschner) [inline .github/workflows/dynamoBinDiff.yml:8]: I have a complication for you - we also need to do this same comparison for the DynamoAll.Net6.Sln targeting windows.

Would you prefer to just duplicate this action for net6 or have this job do 4 builds?

File: .github/scripts/bin_diff.ps1 (hunk 3 above)
Comment (zeusongit) [reply .github/scripts/bin_diff.ps1:242]: 1. Agreed, do you suggest removing modified entries? I think we can let them be.
2. I have added colors to make them pop, green for Added, and red for Deleted entries.
3. Yes, deleted files has a separate "Deleted" label. attaching ss below.
Screenshot 2023-05-22 at 7 22 26 PM

File: .github/scripts/bin_diff.ps1 (hunk 2 above)
Comment (zeusongit) [reply .github/scripts/bin_diff.ps1]: Done

File: .github/scripts/bin_diff.ps1 (hunk 1 above)
Comment (zeusongit) [reply .github/scripts/bin_diff.ps1:1]: Added source.

File: .github/workflows/dynamoBinDiff.yml (hunk 4 above)
Comment (zeusongit) [reply .github/workflows/dynamoBinDiff.yml:8]: Added 2 more builds for Net6-Windows (parallel builds for quick results)

Screenshot 2023-05-22 at 8 10 43 PM

**Review by mjkkirschner — APPROVED** [review]

File: .github/scripts/bin_diff.ps1 (hunk 3 above)
Comment (zeusongit) [reply .github/scripts/bin_diff.ps1:242]: removed hash and side indicator columns as they were not useful.

Screenshot 2023-05-22 at 8 32 56 PM

**Review by QilongTang — APPROVED** [review]

**Comment by zeusongit** [comment]
@mjkkirschner I don't think the coloring worked in Github Action console logs, but it did work in powershell.

**Pull request merged**